src/codemagic/models/xctests/converter.py

```diff
@@ -140,7 +174,169 @@ def actions_invocation_record_to_junit(cls, actions_invocation_record: ActionsIn
+            return None
+
+        failure_messages_nodes = cls._iter_nodes(xc_test_case, XcTestNodeType.FAILURE_MESSAGE)
+        failure_messages = [node.name for node in failure_messages_nodes if node.name]
+        return Error(
+            message=failure_messages[0] if failure_messages else "",
+            type="Error" if any("caught error" in m for m in failure_messages) else "Failure",
+            error_description="\n".join(failure_messages) if len(failure_messages) > 1 else None,
+        )
+
+    @classmethod
+    def _get_test_case_skipped(cls, xc_test_case: XcTestNode) -> Optional[Skipped]:
+        if xc_test_case.result is not XcTestResult.SKIPPED:
+            return None
+
+        failure_messages_nodes = cls._iter_nodes(xc_test_case, XcTestNodeType.FAILURE_MESSAGE)
+        skipped_message_nodes = (node for node in failure_messages_nodes if node.result is XcTestResult.SKIPPED)
+        skipped_messages = [node.name for node in skipped_message_nodes if node.name]
+
+        return Skipped(message="\n".join(skipped_messages))
+
+    @classmethod
+    def _get_test_node_duration(cls, xc_test_case: XcTestNode) -> float:
+        if not xc_test_case.duration:
+            return 0.0
+
+        duration = xc_test_case.duration.replace(",", ".")
+        if duration.endswith("s"):
+            duration = duration[:-1]
+        return float(duration)
+
+    @classmethod
+    def _get_test_case(cls, xc_test_case: XcTestNode, xc_test_suite: XcTestNode) -> TestCase:
+        if xc_test_case.name:
+            method_name = xc_test_case.name
+        elif xc_test_case.node_identifier:
+            method_name = xc_test_case.node_identifier.split("/")[-1]
+        else:
+            method_name = ""
+
+        if xc_test_case.node_identifier:
+            classname = xc_test_case.node_identifier.split("/", maxsplit=1)[0]
+        elif xc_test_suite.name:
+            classname = xc_test_suite.name
+        else:
+            classname = ""
+
+        return TestCase(
+            name=method_name,
+            classname=classname,
+            error=cls._get_test_case_error(xc_test_case),
+            time=cls._get_test_node_duration(xc_test_case),
+            status=xc_test_case.result.value if xc_test_case.result else None,
+            skipped=cls._get_test_case_skipped(xc_test_case),
+        )
+
+    @classmethod
+    def _get_test_suite_properties(
+        cls,
+        xc_test_suite: XcTestNode,
+        xc_test_result_summary: XcSummary,
+    ) -> List[Property]:
+        device = cls._get_run_destination(xc_test_suite)
+
+        properties: List[Property] = [Property(name="title", value=xc_test_suite.name)]
+        if xc_test_result_summary.start_time:
+            Property(name="started_time", value=cls._timestamp(xc_test_result_summary.start_time))
+        if xc_test_result_summary.finish_time:
+            Property(name="ended_time", value=cls._timestamp(xc_test_result_summary.finish_time))
```

Comment:
`properties.append` seems missed here

Reply:
Yes, you're very much correct on that one.

Reply:
Fixed in [d115c03](https://github.com/codemagic-ci-cd/cli-tools/pull/431/commits/d115c032e3e5cbfeea766bf48a25a80e982527d5).